logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/MessageSender.kt

```diff
@@ -73,19 +79,24 @@ class MessageSenderImpl(
                     sessionEstablisher.prepareRecipientsForNewOutgoingMessage(recipients).map { recipients }
                 }.flatMap { recipients ->
                     messageEnvelopeCreator.createOutgoingEnvelope(recipients, message).flatMap { envelope ->
-                        trySendingEnvelopeRetryingIfPossible(conversationId, envelope, message)
+                        trySendingProteusEnvelope(conversationId, envelope, message)
                     }
                 }
         }
 
-    private suspend fun attemptToSendWithMLS(conversationId: ConversationId, groupId: String, message: Message): Either<CoreFailure, Unit> = suspending {
-        mlsMessageCreator.createOutgoingMLSMessage(groupId, message).flatMap { mlsMessage ->
-            // TODO handle mls-stale-message
-            messageRepository.sendMLSMessage(conversationId, mlsMessage)
+    private suspend fun attemptToSendWithMLS(conversationId: ConversationId, groupId: String, message: Message): Either<CoreFailure, Unit> =
+        suspending {
+            mlsMessageCreator.createOutgoingMLSMessage(groupId, message).flatMap { mlsMessage ->
+                // TODO handle mls-stale-message
+                messageRepository.sendMLSMessage(conversationId, mlsMessage)
+            }
         }
-    }
 
-    private suspend fun trySendingEnvelopeRetryingIfPossible(
+    /**
+     * Attempts to send a Proteus envelope
+     * Will handle the failure and retry in case of [SendMessageFailure.ClientsHaveChanged]
+     */
+    private suspend fun trySendingProteusEnvelope(
```

Comment:
This renaming is intentional, as it's related to this PR.

The `retryingIfPossible` was misleading as it only retries in the case of `Proteus` failure, not in other cases.
It also doesn't handle `MLS` sending.